AsyncDisplayKit/ASCellNode.mm

```diff
@@ -120,10 +121,10 @@ - (void)setLayerBacked:(BOOL)layerBacked
   ASDisplayNodeAssert(!layerBacked, @"ASCellNode does not support layer-backing.");
 }
 
-- (void)setNeedsLayout
+- (void)__setNeedsLayout
```

Comment:
Yeah it's intended. Consider, if they call `-setNeedsDisplay` off-main, then basically nothing will happen during `[super setNeedsLayout]` so `self.calculatedSize` won't change. Then at some point in the future, when the base class implementation actually does the remeasure, the cell node won't be in the loop anymore.